src/datasources/asset/AssetDataSource.ts

```diff
@@ -28,22 +46,31 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
   baseUrl = this.instanceSettings.url + '/niapm/v1';
 
   defaultQuery = {
+    type: AssetQueryType.Metadata,
     workspace: '',
+    assetIdentifiers: [],
     minionIds: [],
-    groupBy: []
+    entityType: EntityType.Asset,
   };
 
-  async runQuery(query: AssetMetadataQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
-    return this.processMetadataQuery(query as AssetMetadataQuery);
+  async runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+    switch (query.type) {
+      case AssetQueryType.Metadata:
+        return await this.handleMetadataQuery(query as AssetMetadataQuery);
+      case AssetQueryType.Utilization:
+        return await this.handleUtilizationQuery(query as AssetUtilizationQuery, options);
```

Comment:
@TigranVardanyan 

I mentioned in one of my comments that we should **extract** the asset utilization code in a different plugin. Let's not mix the Asset Metadata code with the Asset Utilization. This way we can have better functionality grouping and control over what features get to our customers moving forward. 

My suggestion on how the plugins should be grouped are as following:

**Assets Utilization plugin** - should contain the code you have in this PR that are related to **Utilization grid**
**Assets Metadata plugin** - should contain the **Assets Metadata grid** and **Assets Summary** (a general overview for the assets)
**Asset Calibration plugin** - should contain the **Calibration forecast** + **Pending Calibration Asset grid** (only calibration related functionality)

The above splitting can ensure the following:

1) You can release the utilization only to certain customers (Since it leverages a beta endpoint).
2) The Metadata plugins can be further developed by other teams to add additional functionalities (such as Asset Summary)
3) The code is less entangled and easier to maintain
4) In case something breaks, it will be scoped only to a single plugin.

If you have further questions or you have any concerns, please reach out to me.

Reply:
@kkerezsi it wasn't me who decided to keep them together
@joshuaprewitt @adamarnesen @TJ-G @kkerezsi @mure  come to a final decision and tell me how to do it
@markacamps FYI

Reply:
@kkerezsi It was decided last week by POs and PMs that the long-term goal is one assets plugin, including the calibration forecast functionality. It's okay to have it split out while it's still in development, but at some point we'll need to merge it back together.